src/Yesod/Auth/OAuth2/Auth0.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE OverloadedStrings #-}
+-- |
+-- OAuth2 plugin for <https://auth0.com>
+--
+-- * Authenticates against specific auth0 tenant
+-- * Uses Auth0 user id (a.k.a [sub](https://auth0.com/docs/api/authentication#get-user-info)) as credentials identifier
+--
+module Yesod.Auth.OAuth2.Auth0
+  (oauth2Auth0HostScopes
+  , defaultAuth0Scopes) where
+
+import Data.Aeson as Aeson
+import qualified Data.Text as T
+import Yesod.Auth.OAuth2.Prelude
+import Prelude
+
+-- | https://auth0.com/docs/api/authentication#get-user-info
+newtype User = User T.Text
+
+instance FromJSON User where
+  parseJSON = withObject "User" $ \o ->
+    User <$> o .: "sub"
+
+-- | https://auth0.com/docs/get-started/apis/scopes/openid-connect-scopes#standard-claims
+defaultAuth0Scopes :: [Text]
+defaultAuth0Scopes = ["openid"]
+
+pluginName :: Text
+pluginName = "auth0"
+
+oauth2Auth0HostScopes :: YesodAuth m => URI -> [Text] -> Text -> Text -> AuthPlugin m
+oauth2Auth0HostScopes host scopes clientId clientSecret =
```

Comment:
- I did test in my sample app (which in private for now but I can make a public one for sure).
- since auth0 is tenant based instead of global (like other oauth2 provider), we have to specific a tenant based url. e.g. `freizl.auth0.com` which only me have admin permission to create App (client_id, client_secret). instead of those oauth2 provider support global tenant, anyone can go to sign-up and create App.
-

Reply:
- here is how it looks like if we adding to example
https://github.com/hw202207/yesod-auth-oauth2/commit/bb319767586b818e01adcae21819972c68fa121a
- one idea I'd think of (not really ideal) is to change `loadPlugin` method - say when it is Auth0, loads `host` from env file besides client id and secret.

Reply:
Gotcha, yeah pulling an `AUTH0_HOST` from env is what I was picturing. You could do it without changing `loadPlugin`.

Something like,

```hs
    loadEnv

    auth0host <- getEnv "AUTH0_HOST"
    appHttpManager <- newManager tlsManagerSettings
    appAuthPlugins <- sequence
        -- When Providers are added, add them here and update .env.example.
        -- Nothing else should need changing.
        --
        -- FIXME: oauth2BattleNet is quite annoying!
        --
        [ loadPlugin oauth2AzureAD "AZURE_AD"
        , loadPlugin (oauth2Auth0Host auth0Host)  "AUTH0"
```